src/Computation/Values/Select.cs

```diff
@@ -47,10 +47,11 @@ public FPExpr AsFloat(IContext context)
 
     public static IValue Create(IValue predicate, IValue trueValue, IValue falseValue)
     {
-        return predicate is IConstantValue p
-            ? p.AsBool()
-                ? trueValue
-                : falseValue
-            : new Select(predicate, trueValue, falseValue);
+        return (predicate, trueValue, falseValue) switch
+        {
+            (IConstantValue p, _, _) => p.AsBool() ? trueValue : falseValue,
+            (_, IConstantValue t, IConstantValue f) when t.AsUnsigned() == f.AsUnsigned() => t,
```

Comment:
This `==` is wrong. It's comparing the instances of `ConstantUnsigned` which don't implement `IEquatable`, rather than their `BigInteger` values.

Reply:
Sorry, I should have caught this. I have even been careful to avoid this myself. Just wasn't paying enough attention.